python/rascaline/utils.py

```diff
@@ -20,3 +22,225 @@ def _call_with_growing_buffer(callback, initial=1024):
             else:
                 raise
     return buffer.value.decode("utf8")
+
+
+def convert_old_hyperparameter_names(hyperparameters, mode="librascal"):
+    """
+    Function to convert old hyperparameter names to those
+    used in rascaline. This function is meant to be dep-
+    recated as rascaline becomes more mainstream, but will
+    serve to help users convert existing workflows.
+
+    Notes
+    -----
+    - This function does validate the values in the hyperparameter
+    dictionary, and it is up to the user to check that they pass
+    valid entries to `rascaline`.
+    - Not all of these parameters are supported in rascaline,
+    and some will raise warnings.
+
+    Parameters
+    ----------
+
+    mode: string in ["librascal", "dscribe"]
+          We anticipate future support for mode=="quip" as well
+
+    hyperparameters: dictionary of hyperparameter keys and values.
+        For mode = `librascal`, the anticipated values are:
+            - `coefficient_subselection`
+            - `compute_gradients`
+            - `covariant_lambda`
+            - `cutoff_function_parameters`
+            - `cutoff_function_type`
+            - `cutoff_smooth_width`
+            - `expansion_by_species_method`
+            - `gaussian_sigma_constant`
+            - `gaussian_sigma_type`
+            - `global_species`
+            - `interaction_cutoff`
+            - `inversion_symmetry`
+            - `max_angular`
+            - `max_radial`
+            - `normalize`
+            - `optimization_args`
+            - `optimization`
+            - `radial_basis`
+            - `soap_type`
+        For mode = `dscribe`, the anticipated values are:
+            - `average`
+            - `crossover`
+            - `dtype`
+            - `nmax`
+            - `lmax`
+            - `periodic`
+            - `rbf`
+            - `rcut`
+            - `sigma`
+            - `sparse`
+            - `species`
+
+    """
+    new_hypers = {}
+
+    if mode == "librascal":
```

Comment:
for readability & extensibility, I would have separate functions for each mode: `_translate_soap_hyperparameters_from_librascal`, `_translate_soap_hyperparameters_from_dscribe` and potentially later `_translate_soap_hyperparameters_from_quip`